logging/servicelogger.go

```diff
@@ -0,0 +1,172 @@
+// when log level is ERROR or higher.
+func (sl *ServiceLogger) Errorln(v ...interface{}) {
+	sl.doPrintln("ERROR", v...)
+}
+
+// Errorf is equivalent to Printf with "ERROR: SERVICENAME: " prepended. Only output
+// when log level is ERROR or higher.
+func (sl *ServiceLogger) Errorf(format string, v ...interface{}) {
+	sl.doPrintf("ERROR", format, v...)
+}
+
+// Critical is equivalent to Print with "CRITICAL: SERVICENAME: " prepended. Only output
+// when log level is CRITICAL.
+func (sl *ServiceLogger) Critical(v ...interface{}) {
+	sl.doPrint("CRITICAL", v...)
+}
+
+// Criticalln is equivalent to Println with "CRITICAL: SERVICENAME: " prepended. Only output
+// when log level is CRITICAL or higher.
+func (sl *ServiceLogger) Criticalln(v ...interface{}) {
+	sl.doPrintln("CRITICAL", v...)
+}
+
+// Criticalf is equivalent to Printf with "CRITICAL: SERVICENAME: " prepended. Only output
+// when log level is CRITICAL or higher.
+func (sl *ServiceLogger) Criticalf(format string, v ...interface{}) {
+	sl.doPrintf("CRITICAL", format, v...)
+}
+
+// Print with "SERVICENAME: " prepended. Only output when log level is SERVICE or higher.
+func (sl *ServiceLogger) Print(v ...interface{}) {
+	sl.doPrint("SERVICE", v...)
+}
+
+// Println with "SERVICENAME: " prepended. Only output when log level is SERVICE or higher.
+func (sl *ServiceLogger) Println(v ...interface{}) {
+	sl.doPrintln("SERVICE", v...)
+}
+
+// Printf with "SERVICENAME: " prepended. Only output when log level is SERVICE or higher.
+func (sl *ServiceLogger) Printf(format string, v ...interface{}) {
+	sl.doPrintf("SERVICE", format, v...)
+}
+
+// prefixString returns "LEVEL: SERVICENAME: " unless the level is at or below
+// SERVICE in which case the prefix is just "SERVICENAME: "
+func (sl *ServiceLogger) prefixString(level string) string {
+	if levelMap[level] <= levelMap["SERVICE"] {
+		return sl.serviceName + ": "
+	}
+	return level + ": " + sl.serviceName + ": "
+}
+
+// doPrint conditionally prints a message prefixed with the log level and service name.
+// If log level is below level, the message is not output.
+func (sl *ServiceLogger) doPrint(level string, v ...interface{}) {
+	if sl.logLevel >= levelMap[level] {
+		sl.Output(2, sl.prefixString(level)+fmt.Sprint(v...))
+	}
+}
+
+// doPrint conditionally prints a printf formatted message prefixed with the log
+// level and service name. If log level is below level, the message is not output.
+func (sl *ServiceLogger) doPrintf(level string, format string, v ...interface{}) {
+	if sl.logLevel >= levelMap[level] {
+		sl.Output(2, sl.prefixString(level)+fmt.Sprintf(format, v...))
+	}
+}
+
+// doPrint conditionally prints a message prefixed with the log level and service name,
+// followed by a newline. If log level is below level, the message is not output.
+func (sl *ServiceLogger) doPrintln(level string, v ...interface{}) {
+	if sl.logLevel >= levelMap[level] {
+		sl.Output(2, sl.prefixString(level)+fmt.Sprintln(v...))
+	}
+}
```

Comment:
Now that we have this I think its safe/time to delete https://github.com/tozny/utils-go/blob/master/logger.go#L10

Reply:
logger.go is removed as part of this PR (it was moved here and refactored).

Reply:
Also much better comments than me 👍